### PR TITLE
Retry on auth failure

### PR DESCRIPTION
**Pull request by jhoule86**
**WIP**

- Retry connections to Wayfair when an authentication problem happens

### CODE REVIEW

File: src/Core/Api/APIService.php [hunk 1]
```diff
@@ -109,11 +130,34 @@ public function query($query, $method = 'post', $variables = [])
           'Body' => $body_for_logging
         ], 'method' => __METHOD__]);
 
-      return $this->client->call($method, $arguments);
+      $response =  $this->client->call($method, $arguments);
```
Comment (jiafanwu) [inline src/Core/Api/APIService.php:133]: Is the auth failure due to token expired? If so, can we just check if the token expired and refresh the token before firing the call?

File: src/Core/Api/Services/FetchDocumentService.php [hunk 2]
```diff
@@ -34,39 +36,86 @@ class FetchDocumentService extends APIService implements FetchDocumentContract
...
+
     try {
-      if (self::isWayfairAPI($url)) {
-        curl_setopt(
-          $ch, CURLOPT_HTTPHEADER, [
-            'Authorization: ' . $this->authService->generateAuthHeader(),
-            ConfigHelper::WAYFAIR_INTEGRATION_HEADER . ': ' . $this->configHelper->getIntegrationAgentHeader()
-          ]
-        );
+      $ch = curl_init();
```
Comment (jiafanwu) [inline src/Core/Api/Services/FetchDocumentService.php:60]: Can we use some http client e.g guzzle instead of curl call?

Comment (jhoule86) [reply src/Core/Api/Services/FetchDocumentService.php:60]: heh you've wandered into this old draft PR, huh?
See line 55: `// FIXME: this should be using a generic client via an interface, not cURL!`

File: src/Core/Api/APIService.php (hunk 1 above)
Comment (jhoule86) [reply src/Core/Api/APIService.php:133]: The `AuthService` module already handles the normal expiration situation.
This draft PR is over a month old though, so I don't suggest investing too much time in reading it. I think I was just experimenting.

**Comment by jhoule86** [comment]
abandoning this effort for now.

**Pull request closed without merging**